lib/commands/general.js

```diff
@@ -42,6 +42,23 @@ commands.mobileIsToastVisible = async function (opts = {}) {
   });
 };
 
+
+// API to invoke methods defined in Android app
+// Eg: To invoke a single method called foo(), opts will be
+//      [{name:"foo"}]
+// If method takes parameters eg foo(String lol, String bar), opts should be
+//      [{name:"foo", args:["lol","bar"]}]
+// Methods can be chained together with opts looking like
+//      [{name:"foo", args:["lol","bar"]}, {name:"qux"}]
+commands.mobileBackdoor = async function (opts = {}) {
+  opts.forEach(function (value) {
```

Comment:
you could use `assertRequiredOptions` helper

Reply:
check an example above

Reply:
done

Reply:
it won't work this way. `opts` must have a key, which value will be equal to an array. We should verify the existice of such key, for example:

```
// opts = {methods: [{name:"foo", args:["lol","bar"]}, {name:"qux"}]}
return await this.espresso.jwproxy.command(`/appium/execute_mobile/backdoor`, 'POST', assertRequiredOptions(opts, ['methods'])});
```

Reply:
I see it working for me with the array.  The key opts is added to array by driver somewhere:
```

2018-11-28 22:47:45:242 - [debug] [W3C (09010c0a)] Calling AppiumDriver.execute() with args: ["mobile: backdoor",[[{"name":"raiseToast","args":["Lol"]}]],"09010c0a-97dc-4cd4-a51b-2690f0081ae0"]
2018-11-28 22:47:45:247 - [debug] [JSONWP Proxy] Proxying [POST /appium/execute_mobile/backdoor] to [POST http://localhost:4567/session/252c75c3-4afa-4769-ad26-aa7d473c1b93/appium/execute_mobile/backdoor] with body: {"opts":[{"name":"raiseToast","args":["Lol"]}]}
```